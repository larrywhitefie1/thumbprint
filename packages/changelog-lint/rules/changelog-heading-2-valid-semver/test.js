const test = require('../../test-utils');
const rule = require('./');

const valid = [
    '## Unreleased',
    '## 1.3.2 - ',
    '## 1.3.2-alpha.0 - ',
    `## Unreleased

## 1.3.2 - `,
];

const invalid = ['## 1.2 - ', '## 1.a.0 -'];

test({
    rule,
    valid,
    invalid,
});
