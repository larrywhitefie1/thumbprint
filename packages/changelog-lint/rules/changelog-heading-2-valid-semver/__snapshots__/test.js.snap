// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invalid case #0 fails 1`] = `
Object {
  "input": "## 1.2 - ",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 10,
          "line": 1,
          "offset": 9,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "“1.2“ is not a valid SemVer version.",
      "ruleId": "changelog-heading-2-valid-semver",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #1 fails 1`] = `
Object {
  "input": "## 1.a.0 -",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 11,
          "line": 1,
          "offset": 10,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "“1.a.0“ is not a valid SemVer version.",
      "ruleId": "changelog-heading-2-valid-semver",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;
